src/test/java/org/jooq/lambda/SeqTest.java

```diff
@@ -1543,6 +1543,22 @@ public void testIterate() {
     }
 
     @Test
+    public void testIterateWhilePresent() {
+        assertEquals(
+              asList(),
+              Seq.iterateWhilePresent(null, i -> Optional.empty()).toList()
+        );
```

Comment:
I understand your surprise - the proposed solution is somewhat inconsistent. The reason I let user provide `null` as `seed` instead of `Optional.empty()` is:
1. `Optional<T>` is essentialy a wrapper for nullable `T`
2. `Optional` is not supposed to be used as a parameter

I see three solutions:
1. call `Objects.requireNonNull` on `seed` (consequences: the resulting `Seq` is always non-empty)
2. change method signature to take `Optional<T> seed` (consequences: we break the convention)
3. leave it as it is (consequences: we're inconsistent in the use of `Optional`s and nullable values)

But what you propose (that `null` `seed` returns a `Seq` with a `null` value) is - in my opinion - the most incoherent solution of all four. If we adopted this, the **only** way `null` could make it into the resulting `Seq` would be as being its first element (because remaining elements are provided by `Optional`, and `Optional` either stores a non-null value or reports it's empty).